frontend/src/components/module-editor/SideBarModuleOverview.tsx

```diff
@@ -59,36 +64,12 @@ const SideBarModuleOverview = (): React.ReactElement => {
         orderedLessons.map((lesson, index) =>
           state.lessons[focusedLesson] === lesson ? (
             lesson && (
-              <Button
-                key={index}
-                onClick={() => setFocus(index)}
-                variant="unstyled"
-                borderLeftColor="brand.royal"
-                borderLeftWidth="5px"
-                borderRadius="0"
-                bg="background.light"
-                textAlign="left"
-                pl="30px"
-                minH="55px"
-                w="100%"
-              >
-                {lesson.title}
-              </Button>
+              <LessonOptions text={lesson.title} isFocused isHovered={isHovered} index={index} setIsHovered={setIsHovered} setFocus={setFocus}/>                  
             )
           ) : (
-            <Button
-              key={index}
-              onClick={() => setFocus(index)}
-              variant="unstyled"
-              textAlign="left"
-              pl="35px"
-              minH="55px"
-              w="100%"
-            >
-              {lesson.title}
-            </Button>
+            <LessonOptions text={lesson.title} isFocused={false} isHovered={isHovered} index={index} setIsHovered={setIsHovered} setFocus={setFocus}/>
```

Comment:
I think index as a prop should be renamed to key in the first place, then passing lesson id as a key would be solid.

Reply:
The `key` prop in react actually has a [special meaning](https://reactjs.org/docs/lists-and-keys.html#keys) and is required only on the outer element in this case. We'll need to pass both a `key` and `index` prop.

Reply:
But if the outermost element is essentially a wrapper for the core component, you either need to be able to handle the updates made to this components with that key, or you can leave it to the outermost barebones element (the button) to deal with it. This is what I think based on my limited understanding of react though D:

What would we need index for this component?

Reply:
Your intuition is good, but the key prop is really only intended in React for when you're looping over a list. It uses it to associate the list item state with each list item. (For example, when we add the ability to drag and drop lessons, React will use the key to know where the lesson moved to.) That's why it needs to be passed in at the top level (at the component directly inside the map, not nested at all), so that React can find the prop; React is only coded to look for that prop at the top level of nesting.

I think you're correct about the index, actually; it probably won't be necessary once the state is refactored inside the component.

Reply:
My bad, made some edits to my past comment. But I think since the focus can only remain on one button, this prop in some form should still remain.